netket/jax/_scanmap.py

```diff
@@ -0,0 +1,39 @@
+import jax
+
+from jax import linear_util as lu
+from jax.api_util import argnums_partial
+
+# from functools import wraps
+
+
+def scan_accum(f, x, op):
+    res0 = f(jax.tree_map(lambda x: x[0], x))
```

Comment:
@PhilipVinc here f is called once, and then below in the loop. 
```suggestion
    res_init = jax.tree_map(lambda x: jnp.zeros(x.shape), jax.eval_shape(f, jax.tree_map(lambda x: x[0], x)))
```

Reply:
yeah let's try to avoid increasing compilation time, and do everything inside the loop.

Reply:
Thinking about it, the reason I did it this way in the first place was that for a general op the zero element is not necessarily the number 0. 
Computing the first batch first avoids having to specify what the 0 element would be.